psp/src/rand.rs

```diff
@@ -0,0 +1,70 @@
+use rand_core::RngCore;
+use core::mem;
+use crate::sys::{self, SceKernelUtilsMt19937Context};
+
+pub struct Mt19937(SceKernelUtilsMt19937Context);
+
+impl Mt19937 {
+    pub fn new(seed: u32) -> Result<Mt19937, ()> { 
+        let mut context: SceKernelUtilsMt19937Context = unsafe { mem::zeroed() };
+        let ret_val = 
+            unsafe {
+                sys::sceKernelUtilsMt19937Init(
+                    &mut context as *mut SceKernelUtilsMt19937Context,
+                    seed
+                )
+            };
+        if ret_val >= 0 {
+            Ok(Mt19937(context))
+        } else {
+            Err(())
+        }
+    }
+
+    pub fn random(&mut self, buf: &mut [u8]) {
+        for chunk in buf.chunks_mut(4) { 
+            chunk.copy_from_slice(&self.random_u32().to_le_bytes()[..chunk.len()]); 
+        }    
```

Comment:
This does quite a bit of work each iteration. I think it might be worth tightening the loop, but I don't know a clean way to do it...

https://godbolt.org/z/j28Ppg